lib/json/api/relationship.rb

```diff
@@ -22,9 +21,15 @@ def collection?
 
       private
 
+      def check_relationship(relationship_hash)
```

Comment:
This refactoring does indeed make sense but there are other places where it would have to be done as well for consistency, so I'd rather have it happen in a separate PR.